tests/admin_views/test_multidb.py

```diff
@@ -0,0 +1,55 @@
+from unittest.mock import patch
+
+from django.conf.urls import url
+from django.contrib import admin
+from django.contrib.auth.models import User
+from django.db import connections, router
+from django.test import TestCase, override_settings
+
+from .models import Book
+
+
+class Router(object):
+    def __init__(self, target):
+        self.target = target
+
+    def db_for_read(self, model, **hints):
+        return self.target
+
+    def db_for_write(self, model, **hints):
+        return self.target
+
+site = admin.AdminSite(name="test_adminsite")
+site.register(Book)
+
+urlpatterns = [
+    url(r'^admin/', site.urls),
+]
+
+
+@override_settings(ROOT_URLCONF=__name__)
+class MultiDatabaseTest(TestCase):
+    multi_db = True
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.superuser = {}
+        for db in connections:
+            router.routers = [Router(db)]
+            cls.superuser[db] = User.objects.create_superuser(username='admin', password='something',
+                                                              email='test@test.org')
+
+    @patch('django.contrib.admin.options.transaction')
+    def test_save_to_each_db(self, mock):
+        for db in connections:
+            # The router must be set before we login.
+            router.routers = [Router(db)]
+            self.client.force_login(self.superuser[db])
+
+            book_name = 'Foobar: 5th edition'
+            self.client.post('/admin/admin_views/book/add/', {'name': book_name})
+            try:
+                self.assertEqual(mock.atomic.call_args_list[-1][1]['using'], db,
+                                 "The wrong `using` kwarg was passed to django.db.transaction.atomic!")
+            except IndexError:
+                self.fail("django.contrib.admin.options.transaction.atomic was never called!")
```

Comment:
I could, but I think that would also check that any other arguments to atomic (i.e. `savepoint`) are not provided. So I did it this way so as to only check the `using` argument.